sklearn/ensemble/tests/test_iforest.py

```diff
@@ -102,18 +102,9 @@ def test_iforest_error():
     warn_msg = "max_samples will be set to n_samples for estimation"
     with pytest.warns(UserWarning, match=warn_msg):
         IsolationForest(max_samples=1000).fit(X)
-    # note that assert_no_warnings does not apply since it enables a
-    # PendingDeprecationWarning triggered by scipy.sparse's use of
-    # np.matrix. See issue #11251.
-    with pytest.warns(None) as record:
-        IsolationForest(max_samples="auto").fit(X)
-    user_warnings = [each for each in record if issubclass(each.category, UserWarning)]
-    assert not user_warnings
-
-    with pytest.warns(None) as record:
-        IsolationForest(max_samples=np.int64(2)).fit(X)
-    user_warnings = [each for each in record if issubclass(each.category, UserWarning)]
-    assert not user_warnings
+
+    IsolationForest(max_samples="auto").fit(X)
+    IsolationForest(max_samples=np.int64(2)).fit(X)
```

Comment:
Thanks @ShanDeng123. Currently the test asserts that there's no UserWarning. We need to keep doing that.